database-portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,35 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for logging in. */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/plain");
```

Comment:
I have a slight problem with this, and I know that it's due to a somewhat ambiguous example that's given for this project. The example sets the content type as "text/html", and it expects you to put "/login" in an anchor tag, like <a href="/login">click here</a>. This action will create an html page and serve it for you. This is an anachronistic pattern, and very rarely used nowadays, which is probably why you are returning this as "text/plain", and requesting this as text data. But, plain text, especially as html, is not preferred, nor is it easy to maintain. So, I think you have two choices:

1) Go all in on the old school "text/html" and make a link to "/login" (old fashioned)
2) Change the content type to json, with userEmail, loginUrl and logoutUrl, and insert it into html clientside (Still very old fashioned, because now we would just let the framework do this for us, but newer than the other option).